pickhardtpayments/Payment.py

```diff
@@ -0,0 +1,207 @@
+import logging
+import time
+
+import Attempt
+
+
+class Payment:
+    """
+    Payment stores the information about an amount of sats to be delivered from source to destination.
+
+    When sending an amount of sats from sender to receiver, a payment is usually split up and sent across
+    several paths, to increase the probability of being successfully delivered.
+    The PaymentClass holds all necessary information about a payment.
+    It also holds information that helps to calculate performance statistics about the payment.
+
+    :param sender: sender address for the payment
+    :type sender: class:`str`
+    :param receiver: receiver address for the payment
+    :type receiver: class:`str`
+    :param total_amount: The total amount of sats to be delivered from source address to destination address.
+    :type total_amount: class:`int`
+    """
+
+    def __init__(self, sender, receiver, total_amount: int = 1):
+        """Constructor method
+        """
+        self._successful = False
+        self._ppm = -1
+        self._fee = -1
+        self._end_time = -1
+        self._sender = sender
+        self._receiver = receiver
+        self._total_amount = total_amount
+        self._attempts = list()
+        self._start_time = time.time()
+
+    def __str__(self):
+        return "Payment with {} attempts to deliver {} sats from {} to {}".format(len(self._attempts),
+                                                                                  self._total_amount,
+                                                                                  self._sender[-8:],
```

Comment:
why not [:8]  or is that the same?

Reply:
it's not the same. I don't know - a while back I decided to shorten the channel names to its last 8 characters. If there's some convention, then we should of course use that instead